app/design/adminhtml/default/default/template/page/notices.phtml

```diff
@@ -29,5 +29,9 @@
     </noscript>
 <?php endif ?>
 <?php if ($this->displayDemoNotice()): ?>
+    <?php if (strlen($this->getDemoNoticeText())): ?>
+    <p class="demo-notice"><?php echo $this->getDemoNoticeText() ?></p>
+    <?php else: ?>
     <p class="demo-notice"><?php echo $this->__('This is a demo store. Any orders placed through this store will not be honored or fulfilled.') ?></p>
```

Comment:
Why not add the default in `app/code/core/Mage/Page/etc/config.xml`?

```xml
    <default>
        <design>
            <head>
                <demonotice_text>This is a demo store. Any orders placed through this store will not be honored or fulfilled.</demonotice_text>
            </head>
        </design>
    </default>
```

Reply:
Nice idea, but strlen () is still required to not have an empty notice. (?)

Reply:
@kiatng this store default text to db. Do we want it?

Reply:
I suggest checking in the backend only, if the notice is empty; "Display Store Notice is enabled but the text is not set, please set it [here](link)."

Reply:
Thats why im asking .... w/o text set, just fallback to default message, Overhead?

What should happen?

- do not save empty text - forces the user to add something
- save empty text and inform the user

Thanks for input :)

Reply:
May be also always include: "Display Store Notice is enabled, disable it here."

Reply:
It alll  started with a simple idea ...

... you want to make it perfect :)

Note to myself ... addd xml form validation ...

Reply:
I'd also use the default value in XML

Reply:
Textarea is the right approach. The default phrase should be in the XML file. I will test it these days.